### DISCUSSION TITLE
"On" events can block event loop ?

### DISCUSSION THREAD

**Discussion by pas8690** [nodejs/help#4969] [Q&A]
Hi at all, 
imagine that i have this piece of code ( with usage of express module ) : 

```js
app.use("/*", async (req, res, next) => {  
    //some code
    res.on("finish", () => {  
        //example JSON.parse(JSON.stringify(req.body))
    });  
    next();  
});
```

The question is: "The event loop can be blocked If i write a methods that do `JSON.parse()` and `JSON.stringify()` ( or other network calls) inside `on("finish")` block ? This block can be solved with usage of async methods ?
Thank you

**Comment by Trott** [comment]
Yes, `JSON.parse()` will block the event loop if it takes a long time. Event callbacks don't run in a separate event loop.

Using `async` does not change this. `JSON.parse()` will still block in that case.

One possible solution is to run `JSON.parse()` in a worker (`worker_thread` in Node.js, WebWorker in the browser).
